--- Python-2.7.14/setup.py	2017-09-16 12:38:35.000000000 -0500
+++ Python-2.7.14.new/setup.py	2017-12-14 16:06:47.177271008 -0600
@@ -809,8 +809,7 @@
                                libraries=math_libs) )
         # Detect SSL support for the socket module (via _ssl)
         search_for_ssl_incs_in = [
-                              '/usr/local/ssl/include',
-                              '/usr/contrib/ssl/include/'
+                              '/opt/ssl/usr/local/ssl/include'
                              ]
         ssl_incs = find_file('openssl/ssl.h', inc_dirs,
                              search_for_ssl_incs_in
@@ -821,8 +820,7 @@
             if krb5_h:
                 ssl_incs += krb5_h
         ssl_libs = find_library_file(self.compiler, 'ssl',lib_dirs,
-                                     ['/usr/local/ssl/lib',
-                                      '/usr/contrib/ssl/lib/'
+                                     ['/opt/ssl/usr/local/ssl/lib'
                                      ] )
 
         if (ssl_incs is not None and
diff -Naur Python-2.7.14.org/Modules/Setup.dist Python-2.7.14/Modules/Setup.dist
--- Python-2.7.14.org/Modules/Setup.dist	2017-09-16 12:38:35.000000000 -0500
+++ Python-2.7.14/Modules/Setup.dist	2017-12-16 00:00:10.072007650 -0600
@@ -215,10 +215,10 @@
 
 # Socket module helper for SSL support; you must comment out the other
 # socket line above, and possibly edit the SSL variable:
-#SSL=/usr/local/ssl
-#_ssl _ssl.c \
-#	-DUSE_SSL -I$(SSL)/include -I$(SSL)/include/openssl \
-#	-L$(SSL)/lib -lssl -lcrypto
+SSL=/opt/ssl/usr/local/ssl
+_ssl _ssl.c \
+	-DUSE_SSL -I$(SSL)/include -I$(SSL)/include/openssl \
+	-L$(SSL)/lib -lssl -lcrypto
 
 # The crypt module is now disabled by default because it breaks builds
 # on many systems (where -lcrypt is needed), e.g. Linux (I believe).
